### PR TITLE
REF: implement LocationIndexer._tupleize_axis_indexer

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

Trying to roll back some complexity that causes problems with bool-indexers.

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -784,23 +804,18 @@ def _is_nested_tuple_indexer(self, tup: tuple) -> bool:
...
-                    keyidx.append(self._convert_to_indexer(key, axis=axis))
-                else:
-                    keyidx.append(slice(None))
-        else:
-            self._validate_key_length(key)
-            for i, k in enumerate(key):
-                idx = self._convert_to_indexer(k, axis=i)
-                keyidx.append(idx)
+        self._validate_key_length(key)
+        for i, k in enumerate(key):
```
Comment (jreback) [inline pandas/core/indexing.py:812]: could be a list-comprehension

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -640,14 +642,15 @@ def _get_setitem_indexer(self, key):
...
         if self.name == "loc":
+            # always holds here bc iloc overrides _get_setitem_indexer
             self._ensure_listlike_indexer(key)
 
         if isinstance(key, tuple):
             for x in key:
                 check_deprecated_indexers(x)
 
         if self.axis is not None:
-            return self._convert_tuple(key)
```
Comment (jreback) [inline pandas/core/indexing.py:650]: can we kill _convert_tuple?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexing.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/indexing.py:650]: not sure yet; overarching goal is dealing with some trouble involving bool indexers

File: pandas/core/indexing.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexing.py:812]: sure

Comment (jbrockmendel) [reply pandas/core/indexing.py:812]: (will add to the forthcoming "collected follow-ups" branch)

**Review by jreback — APPROVED** [review]

**Pull request merged**